src/idl_gen_cpp.cpp

```diff
@@ -405,12 +405,11 @@ class CppGenerator : public BaseGenerator {
   }
 
   std::string TableCreateSignature(StructDef &struct_def, bool predecl) {
-    return "inline flatbuffers::Offset<" + struct_def.name + "> Create" +
-           struct_def.name  +
-           "(flatbuffers::FlatBufferBuilder &_fbb, const " +
-           NativeName(struct_def.name) +
-           " *_o, const flatbuffers::rehasher_function_t *rehasher" +
-           (predecl ? " = nullptr" : "") + ")";
+    return "template<>\ninline flatbuffers::Offset<" + 
```

Comment:
rather than modifying the existing code (and breaking the current API), please just make the templated version an alternative way of calling the same thing.